code/game/gamemodes/events/space_weather.dm

```diff
@@ -77,10 +77,10 @@
 	endWhen = rand(300, 600)
 
 /datum/event/photon_vortex/start()
-	for(var/obj/item/device/lighting/L in world)
+	for(var/obj/item/device/lighting/L in ship_areas)
 		L.brightness_on = L.brightness_on / 3
 		L.update_icon()
-	for(var/obj/machinery/light/l in world)
+	for(var/obj/machinery/light/l in ship_areas)
```

Comment:
That loop would just do nothing, as `ship_areas` contains areas (duh), not light bulbs.
Here we need two loops. One for areas and nested in it another one, for lights bulbs.
Something like that:
```
for(var/area as anything in ship_areas)
	for(var/obj/item/device/lighting/lightbulb in area)
```

Reply:
i hate byondcode i hate byondcode i hate byondcode i hate byondcode

Reply:
Skill issue, unironically.
BYOND/DM is rather simple (that doesn't make it a good language tho), and you would face similar issue in any other language.

Reply:
clearly it's byond that makes me code poorly